pkg/rpcclient/wsclient.go

```diff
@@ -640,13 +654,13 @@ func (c *WSClient) performSubscription(params []interface{}, rcvr notificationRe
 //
 // Deprecated: please, use ReceiveBlocks. This method will be removed in future versions.
 func (c *WSClient) SubscribeForNewBlocks(primary *int) (string, error) {
-	var flt *neorpc.BlockFilter
+	var flt interface{}
 	if primary != nil {
 		flt = &neorpc.BlockFilter{Primary: primary}
```

Comment:
And if you're to do `neorpc.BlockFilter{Primary: primary}` here, you can avoid changing `Filter()` at all.